airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py

```diff
@@ -758,6 +758,11 @@ def should_validate_schema_fixture(self, inputs: BasicReadTestConfig, test_stric
         else:
             return inputs.validate_schema
 
+    @pytest.fixture(name="should_fail_on_extra_columns")
+    def should_fail_on_extra_columns_fixture(self, inputs: BasicReadTestConfig):
+        # TODO (Ella): enforce this param once all connectors are passing, sunset additionalProperties on high-level object
```

Comment:
I thought we wanted to keep ensuring that additionalProperties was true so we could upgrade the version of a connector and not fail a yet-to-be-migrated config?

Reply:
I agree that we shouldn't make a change about additionalProperties in this PR 😄.  I was reacting to your comment mostly.  I _think_ we can remove the check once we have config migrations, (1) I'm low-confident in that answer and (2) that's not any time soon.

Reply:
Oh yeah no I was saying that half of my comment shouldn't be there (I already removed it) 

I think we can enforce the parameter (all sources should fail on extra columns if they run schema validation), but not actually stop users from putting `additionalProperties: true` in the top level of the spec, for the reasons you mentioned. :) 

Beleive it or not I think we're on the same page haha